test/e2e/utils_test.go

```diff
@@ -120,28 +120,58 @@ func exposeAdminAPI(ctx context.Context, t *testing.T, env environments.Environm
 // returns the modified manifest path. If there is any issue patching the manifest, it will log the issue and return
 // the original provided path.
 func getTestManifest(t *testing.T, baseManifestPath string) (io.Reader, error) {
+	var manifestsReader io.Reader
+	manifestsReader, err := os.Open(baseManifestPath)
+	if err != nil {
+		return nil, err
+	}
+
 	var imagetag string
 	if imageLoad != "" {
 		imagetag = imageLoad
 	} else {
 		imagetag = imageOverride
 	}
-	if imagetag == "" {
-		return os.Open(baseManifestPath)
+
+	if imagetag != "" {
+		split := strings.Split(imagetag, ":")
+		if len(split) < 2 {
+			t.Logf("could not parse override image '%v', using default manifest %v", imagetag, baseManifestPath)
+			return manifestsReader, nil
+		}
+		manifestsReader, err = patchControllerImage(manifestsReader, strings.Join(split[0:len(split)-1], ":"),
+			split[len(split)-1])
+		if err != nil {
+			t.Logf("failed patching override image '%v' (%v), using default manifest %v", imagetag, err, baseManifestPath)
+			return manifestsReader, nil
+		}
 	}
-	split := strings.Split(imagetag, ":")
-	if len(split) < 2 {
-		t.Logf("could not parse override image '%v', using default manifest %v", imagetag, baseManifestPath)
-		return os.Open(baseManifestPath)
+
+	var kongImageTag string
+	if kongImageLoad != "" {
+		kongImageTag = kongImageLoad
+	} else {
+		kongImageTag = kongImageOverride
 	}
-	modified, err := patchControllerImage(baseManifestPath, strings.Join(split[0:len(split)-1], ":"),
-		split[len(split)-1])
-	if err != nil {
-		t.Logf("failed patching override image '%v' (%v), using default manifest %v", imagetag, err, baseManifestPath)
-		return os.Open(baseManifestPath)
+	if kongImageTag != "" {
+		t.Logf("replace kong image to %s", kongImageTag)
+		split := strings.Split(kongImageTag, ":")
+		if len(split) < 2 {
+			t.Logf("could not parse override image '%v', using default manifest %v", imagetag, baseManifestPath)
+			return manifestsReader, nil
+		}
+		manifestsReader, err = patchKongImage(manifestsReader, strings.Join(split[0:len(split)-1], ":"),
+			split[len(split)-1])
```

Comment:
Do you mind naming the variables here? The split and indexing here is a bit magic

Reply:
fixed.